In [1]:
import csv
from datetime import datetime
from collections import Counter

import pandas as pd

import numpy as np
import matplotlib.pyplot as plt

%matplotlib inline

import re

In [106]:
pd.set_option('display.max_columns', 100)
pd.set_option('display.max_rows', 40)
pd.set_option('display.precision', 5)
pd.set_option('expand_frame_repr', True)

In [3]:
df = pd.read_table('Tech_Companies_NYC.csv', sep=',')
df.head()

Company Name Address Address2   City  Category Name  URL
0    "Document Prep-Program"'   ' - '    ' - '  ' - '            NaN  NaN
1  "More than just Figleaves'   ' - '    ' - '  ' - '            NaN  NaN
2                    #Fit4ME'   ' - '    ' - '  ' - '            NaN  NaN
3                 'brellaBox'   ' - '    ' - '  ' - '            NaN  NaN
4                 'wichcraft'   ' - '    ' - '  ' - '            NaN  NaN

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6335 entries, 0 to 6334
Data columns (total 6 columns):
Company Name     6335 non-null object
Address          6320 non-null object
Address2         6298 non-null object
City             6296 non-null object
Category Name    0 non-null float64
URL              0 non-null float64
dtypes: float64(2), object(4)
memory usage: 297.0+ KB


In [5]:
df1=df.replace('-', np.nan )

In [6]:
df1.head()

Company Name Address Address2   City  Category Name  URL
0    "Document Prep-Program"'   ' - '    ' - '  ' - '            NaN  NaN
1  "More than just Figleaves'   ' - '    ' - '  ' - '            NaN  NaN
2                    #Fit4ME'   ' - '    ' - '  ' - '            NaN  NaN
3                 'brellaBox'   ' - '    ' - '  ' - '            NaN  NaN
4                 'wichcraft'   ' - '    ' - '  ' - '            NaN  NaN

In [7]:
df.columns

Index(['Company Name', 'Address', 'Address2', 'City', 'Category Name', 'URL'], dtype='object')

In [8]:
df.Address.value_counts()

' - '                                                5000
'902 Broadway'                                         17
'175 Varick Street'                                    13
'68 Jay Street'                                         8
'20 Jay Street'                                         7
'45 Main Street'                                        7
'560 Broadway'                                          6
'79 Madison Avenue'                                     5
'394 Broadway'                                          5
'1133 Broadway'                                         5
'175 Varick St.'                                        5
'222 Broadway'                                          4
                                                     ... 
'1345 Avenue of the Americas, New York, NY 10105'       1
'540 Atlantic Ave'                                      1
'39 West 13th St'                                       1
'122 West 26th St'                                      1
'466 Lexington

In [9]:
df1 = df.replace(regex=True,to_replace=' - ',value= np.nan)
df1.head()

Company Name Address Address2 City  Category Name  URL
0    "Document Prep-Program"'     NaN      NaN  NaN            NaN  NaN
1  "More than just Figleaves'     NaN      NaN  NaN            NaN  NaN
2                    #Fit4ME'     NaN      NaN  NaN            NaN  NaN
3                 'brellaBox'     NaN      NaN  NaN            NaN  NaN
4                 'wichcraft'     NaN      NaN  NaN            NaN  NaN

In [10]:
df1.Address.value_counts()

'902 Broadway'                                       17
'175 Varick Street'                                  13
'68 Jay Street'                                       8
'45 Main Street'                                      7
'20 Jay Street'                                       7
'560 Broadway'                                        6
'1133 Broadway'                                       5
'79 Madison Avenue'                                   5
'175 Varick St.'                                      5
'394 Broadway'                                        5
'222 Broadway'                                        4
'594 Broadway'                                        4
                                                     ..
'1345 Avenue of the Americas, New York, NY 10105'     1
'540 Atlantic Ave'                                    1
'39 West 13th St'                                     1
'122 West 26th St'                                    1
'466 Lexington Ave, # 6'                        

In [11]:
df1.Address2.value_counts()

'4th Floor'                            39
'5th Floor'                            25
'3rd Floor'                            24
'2nd Floor'                            22
'11th Floor'                           14
'6th Floor'                            14
'7th Floor'                            10
'8th Floor'                            10
'10th Floor'                            8
'4th floor'                             7
'9th Floor'                             7
'14th Floor'                            6
                                       ..
'2nd Fl'                                1
'Suite 227D'                            1
'suite 1024'                            1
'Suite 70A'                             1
'World Trade Center Tower 7, fl 46'     1
'Suite #7'                              1
'4th FL'                                1
'Suite 1522'                            1
'7th Floor '                            1
'18th Floor/Suite 1839'                 1
'Suite 7A'                        

In [12]:
df1.City.value_counts()

'New York'                               1642
'Brooklyn'                                230
'Long Island City'                         15
'Staten Island'                            13
'Bronx'                                     9
'Queens'                                    6
'New York City'                             5
'New York '                                 4
'NY'                                        2
'Rosedale'                                  1
'Glen Cove'                                 1
'Greenvale'                                 1
                                         ... 
'http://devbootcamp.com/location-nyc/       1
'Forest Hills, Queens'                      1
'Manhattan'                                 1
'Forest Hills'                              1
'Great Neck'                                1
'sproutvideo.com                            1
'NEW YORK'                                  1
'bronx'                                     1
'New'                             

In [18]:
df1.Address.unique()

array([nan, "'1239 Broadway'", "'121 Varick Street'", ...,
       "'1 Liberty Plaza'", "'173 Avenue A '", "'1116 east 224th street'"], dtype=object)

In [19]:
df1.Address.nunique()

1120

In [20]:
df2 = df1[df1.Address.notnull()]

In [21]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1320 entries, 9 to 6322
Data columns (total 6 columns):
Company Name     1319 non-null object
Address          1320 non-null object
Address2         700 non-null object
City             1262 non-null object
Category Name    0 non-null float64
URL              0 non-null float64
dtypes: float64(2), object(4)
memory usage: 72.2+ KB


In [22]:
df2.head()

Company Name                                      Address  \
9       10 Speed Labs'                              '1239 Broadway'   
17  1800Postcards.com'                          '121 Varick Street'   
20          1938 News'                                 '1 Astor Pl'   
21            1DocWay'  '483 Broadway, Floor 2, New York, NY 10013'   
27        1stdibs.com'                             '51 Astor Place'   

                   Address2        City  Category Name  URL  
9               'Penthouse'  'New York'            NaN  NaN  
17              '4th Floor'  'New York'            NaN  NaN  
20                      NaN  'New York'            NaN  NaN  
21  '483 Broadway, Floor 2'  'New York'            NaN  NaN  
27            'Third Floor'  'New York'            NaN  NaN

In [23]:
df2.Address.value_counts()

'902 Broadway'                                       17
'175 Varick Street'                                  13
'68 Jay Street'                                       8
'45 Main Street'                                      7
'20 Jay Street'                                       7
'560 Broadway'                                        6
'79 Madison Avenue'                                   5
'175 Varick St.'                                      5
'1133 Broadway'                                       5
'394 Broadway'                                        5
'636 Broadway'                                        4
'122 West 26th Street'                                4
                                                     ..
'1345 Avenue of the Americas, New York, NY 10105'     1
'540 Atlantic Ave'                                    1
'39 West 13th St'                                     1
'122 West 26th St'                                    1
'466 Lexington Ave, # 6'                        

### Searching sub string of Address

In [25]:
df2[(df2['Address'].str.contains('Broadway', re.IGNORECASE))]

Company Name  \
9                         10 Speed Labs'   
21                              1DocWay'   
52         4C Insight (The Echo System)'   
68                                 87AM'   
73                              9mmedia'   
90              ACE Innovative Networks'   
112                                 AOL'   
124                             AUDUBON'   
132                           About.com'   
136                             Acadaca'   
175               Adtivity by Appssavvy'   
182                              Adweek'   
...                                  ...   
5569                             Vitals'   
5576                              Viyet'   
5591                              Voxel'   
5592                               Voxy'   
5660                          Wednesday'   
5864                             ZocDoc'   
5869                         Zola, Inc.'   
5876                         activecell'   
5976  dotbox (acquired by MDC Partners)'   
5989                        eDiningNews'   
6145                        mohchi inc.'   
6273                              tracx'   

                                              Address  \
9                                     '1239 Broadway'   
21        '483 Broadway, Floor 2, New York, NY 10013'   
52                                    '1133 Broadway'   
68                                      '61 Broadway'   
73                                     '627 Broadway'   
90                                     '277 Broadway'   
112                '770 Broadway, New York, NY 10003'   
124                                   '3960 Broadway'   
132    '1500 Broadway, 6th Floor, New York, NY 10016'   
136                               '636 Broadway #516'   
175                                    '594 Broadway'   
182                                    '770 Broadway'   
...                                               ...   
5569                                   '632 Broadway'   
5576                                   '584 Broadway'   
5591                                    '29 Broadway'   
5592                                   '632 Broadway'   
5660                                   '530 Broadway'   
5864                                   '568 Broadway'   
5869                         '394 Broadway 3rd Floor'   
5876                                   '902 Broadway'   
5976                           '853 Broadway, 4th FL'   
5989                                   '394 Broadway'   
6145  '1601 Broadway, 12th Floor, New York, NY 10019'   
6273                           '1115 Broadway Street'   

                     Address2        City  Category Name  URL  
9                 'Penthouse'  'New York'            NaN  NaN  
21    '483 Broadway, Floor 2'  'New York'            NaN  NaN  
52               'Suite 1420'  'New York'            NaN  NaN  
68                        NaN  'New York'            NaN  NaN  
73                        NaN  'New York'            NaN  NaN  
90                      '807'  'New York'            NaN  NaN  
112                       NaN         NaN            NaN  NaN  
124                       NaN  'New York'            NaN  NaN  
132                       NaN         NaN            NaN  NaN  
136                       NaN  'New York'            NaN  NaN  
175               'Suite 207'  'New York'            NaN  NaN  
182               '7th floor'  'New York'            NaN  NaN  
...                       ...         ...            ...  ...  
5569                    '301'  'New York'            NaN  NaN  
5576              'Suite 408'  'New York'            NaN  NaN  
5591             '30th Floor'  'New York'            NaN  NaN  
5592                      NaN  'New York'            NaN  NaN  
5660              '6th Floor'  'New York'            NaN  NaN  
5864              '"9th Floor         NaN            NaN  NaN  
5869                      NaN  'New York'            NaN  NaN  
5876                'Floor 4'  'New York'            NaN

### Splitting & Cleaning Address 

In [26]:
df2['Address_Num'],df2['Address_StName'] = df2['Address'].str.split(' ', 1).str
df2.head()

/Users/sdutta/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


Company Name                                      Address  \
9       10 Speed Labs'                              '1239 Broadway'   
17  1800Postcards.com'                          '121 Varick Street'   
20          1938 News'                                 '1 Astor Pl'   
21            1DocWay'  '483 Broadway, Floor 2, New York, NY 10013'   
27        1stdibs.com'                             '51 Astor Place'   

                   Address2        City  Category Name  URL Address_Num  \
9               'Penthouse'  'New York'            NaN  NaN       '1239   
17              '4th Floor'  'New York'            NaN  NaN        '121   
20                      NaN  'New York'            NaN  NaN          '1   
21  '483 Broadway, Floor 2'  'New York'            NaN  NaN        '483   
27            'Third Floor'  'New York'            NaN  NaN         '51   

                            Address_StName  
9                                Broadway'  
17                          Varick Street'  
20                               Astor Pl'  
21  Broadway, Floor 2, New York, NY 10013'  
27                            Astor Place'

In [27]:
df2['Address_Num'] = df2['Address_Num'].str.replace(r"'","")

df2.head()

/Users/sdutta/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


Company Name                                      Address  \
9       10 Speed Labs'                              '1239 Broadway'   
17  1800Postcards.com'                          '121 Varick Street'   
20          1938 News'                                 '1 Astor Pl'   
21            1DocWay'  '483 Broadway, Floor 2, New York, NY 10013'   
27        1stdibs.com'                             '51 Astor Place'   

                   Address2        City  Category Name  URL Address_Num  \
9               'Penthouse'  'New York'            NaN  NaN        1239   
17              '4th Floor'  'New York'            NaN  NaN         121   
20                      NaN  'New York'            NaN  NaN           1   
21  '483 Broadway, Floor 2'  'New York'            NaN  NaN         483   
27            'Third Floor'  'New York'            NaN  NaN          51   

                            Address_StName  
9                                Broadway'  
17                          Varick Street'  
20                               Astor Pl'  
21  Broadway, Floor 2, New York, NY 10013'  
27                            Astor Place'

In [28]:
df2['Address_StName'].replace(regex=True,inplace=True,to_replace=r"'",value=r'')
df2.head()

/Users/sdutta/anaconda/lib/python3.6/site-packages/pandas/core/generic.py:3813: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)


Company Name                                      Address  \
9       10 Speed Labs'                              '1239 Broadway'   
17  1800Postcards.com'                          '121 Varick Street'   
20          1938 News'                                 '1 Astor Pl'   
21            1DocWay'  '483 Broadway, Floor 2, New York, NY 10013'   
27        1stdibs.com'                             '51 Astor Place'   

                   Address2        City  Category Name  URL Address_Num  \
9               'Penthouse'  'New York'            NaN  NaN        1239   
17              '4th Floor'  'New York'            NaN  NaN         121   
20                      NaN  'New York'            NaN  NaN           1   
21  '483 Broadway, Floor 2'  'New York'            NaN  NaN         483   
27            'Third Floor'  'New York'            NaN  NaN          51   

                           Address_StName  
9                                Broadway  
17                          Varick Street  
20                               Astor Pl  
21  Broadway, Floor 2, New York, NY 10013  
27                            Astor Place

In [30]:
df2['Address_St.Name'],df2['X'] = df2['Address_StName'].str.split(',', 1).str

df2.head()

/Users/sdutta/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


Company Name                                      Address  \
9       10 Speed Labs'                              '1239 Broadway'   
17  1800Postcards.com'                          '121 Varick Street'   
20          1938 News'                                 '1 Astor Pl'   
21            1DocWay'  '483 Broadway, Floor 2, New York, NY 10013'   
27        1stdibs.com'                             '51 Astor Place'   

                   Address2        City  Category Name  URL Address_Num  \
9               'Penthouse'  'New York'            NaN  NaN        1239   
17              '4th Floor'  'New York'            NaN  NaN         121   
20                      NaN  'New York'            NaN  NaN           1   
21  '483 Broadway, Floor 2'  'New York'            NaN  NaN         483   
27            'Third Floor'  'New York'            NaN  NaN          51   

                           Address_StName Address_St.Name  \
9                                Broadway        Broadway   
17                          Varick Street   Varick Street   
20                               Astor Pl        Astor Pl   
21  Broadway, Floor 2, New York, NY 10013        Broadway   
27                            Astor Place     Astor Place   

                               X  
9                            NaN  
17                           NaN  
20                           NaN  
21   Floor 2, New York, NY 10013  
27                           NaN

In [31]:
new_df = df2[['Company Name', 'Address', 'Address2','City','Address_Num','Address_St.Name']].copy()
new_df.head()

Company Name                                      Address  \
9       10 Speed Labs'                              '1239 Broadway'   
17  1800Postcards.com'                          '121 Varick Street'   
20          1938 News'                                 '1 Astor Pl'   
21            1DocWay'  '483 Broadway, Floor 2, New York, NY 10013'   
27        1stdibs.com'                             '51 Astor Place'   

                   Address2        City Address_Num Address_St.Name  
9               'Penthouse'  'New York'        1239        Broadway  
17              '4th Floor'  'New York'         121   Varick Street  
20                      NaN  'New York'           1        Astor Pl  
21  '483 Broadway, Floor 2'  'New York'         483        Broadway  
27            'Third Floor'  'New York'          51     Astor Place

In [32]:
new_df.rename(columns={'Address_St.Name': 'Add_StName'},inplace=True)
new_df.head()

Company Name                                      Address  \
9       10 Speed Labs'                              '1239 Broadway'   
17  1800Postcards.com'                          '121 Varick Street'   
20          1938 News'                                 '1 Astor Pl'   
21            1DocWay'  '483 Broadway, Floor 2, New York, NY 10013'   
27        1stdibs.com'                             '51 Astor Place'   

                   Address2        City Address_Num     Add_StName  
9               'Penthouse'  'New York'        1239       Broadway  
17              '4th Floor'  'New York'         121  Varick Street  
20                      NaN  'New York'           1       Astor Pl  
21  '483 Broadway, Floor 2'  'New York'         483       Broadway  
27            'Third Floor'  'New York'          51    Astor Place

In [33]:
new_df.Add_StName.value_counts()

Broadway                       165
Madison Avenue                  25
Fifth Avenue                    24
Varick Street                   21
Park Avenue South               20
Jay Street                      17
Avenue of the Americas          17
5th Avenue                      16
5th Ave                         10
Madison Ave                     10
Water Street                    10
Third Avenue                     9
                              ... 
26th avenue                      1
W26th Street                     1
East 11th Street 11th Floor      1
Morton Street                    1
Broad st.                        1
W 86TH ST                        1
W. 30th St                       1
East 92nd Street                 1
Piermont Plaza                   1
23rd Ave                         1
State Street 19th Floor          1
W 57th Street                    1
Name: Add_StName, Length: 700, dtype: int64

In [34]:
new_df[(new_df['Add_StName'].str.contains('Broadway', re.IGNORECASE, na=False))]

Company Name  \
9                         10 Speed Labs'   
21                              1DocWay'   
52         4C Insight (The Echo System)'   
68                                 87AM'   
73                              9mmedia'   
90              ACE Innovative Networks'   
112                                 AOL'   
124                             AUDUBON'   
132                           About.com'   
136                             Acadaca'   
175               Adtivity by Appssavvy'   
182                              Adweek'   
...                                  ...   
5569                             Vitals'   
5576                              Viyet'   
5591                              Voxel'   
5592                               Voxy'   
5660                          Wednesday'   
5864                             ZocDoc'   
5869                         Zola, Inc.'   
5876                         activecell'   
5976  dotbox (acquired by MDC Partners)'   
5989                        eDiningNews'   
6145                        mohchi inc.'   
6273                              tracx'   

                                              Address  \
9                                     '1239 Broadway'   
21        '483 Broadway, Floor 2, New York, NY 10013'   
52                                    '1133 Broadway'   
68                                      '61 Broadway'   
73                                     '627 Broadway'   
90                                     '277 Broadway'   
112                '770 Broadway, New York, NY 10003'   
124                                   '3960 Broadway'   
132    '1500 Broadway, 6th Floor, New York, NY 10016'   
136                               '636 Broadway #516'   
175                                    '594 Broadway'   
182                                    '770 Broadway'   
...                                               ...   
5569                                   '632 Broadway'   
5576                                   '584 Broadway'   
5591                                    '29 Broadway'   
5592                                   '632 Broadway'   
5660                                   '530 Broadway'   
5864                                   '568 Broadway'   
5869                         '394 Broadway 3rd Floor'   
5876                                   '902 Broadway'   
5976                           '853 Broadway, 4th FL'   
5989                                   '394 Broadway'   
6145  '1601 Broadway, 12th Floor, New York, NY 10019'   
6273                           '1115 Broadway Street'   

                     Address2        City Address_Num          Add_StName  
9                 'Penthouse'  'New York'        1239            Broadway  
21    '483 Broadway, Floor 2'  'New York'         483            Broadway  
52               'Suite 1420'  'New York'        1133            Broadway  
68                        NaN  'New York'          61            Broadway  
73                        NaN  'New York'         627            Broadway  
90                      '807'  'New York'         277            Broadway  
112                       NaN         NaN         770            Broadway  
124                       NaN  'New York'        3960            Broadway  
132                       NaN         NaN        1500            Broadway  
136                       NaN  'New York'         636       Broadway #516  
175               'Suite 207'  'New York'         594            Broadway  
182               '7th floor'  'New York'         770            Broadway  
...                       ...         ...         ...                 ...  
5569                    '301'  'New York'         632            Broadway  
5576              'Suite 408'  'New York'         584            Broadway  
5591             '30th Floor'  'New York'          29            Broadway  
5592                      NaN  'New York'         632            Broadway  
5660              '6th Floor'  '

In [45]:
gb = new_df.sort_values(by = ['Add_StName'], ascending=True)
gb[:30]

Company Name               Address      Address2  \
4978                     Tapjoy, Inc'  '166  Mercer Street'     'Floor 2'   
2171                   Group 8A, LLC'       '475 10th Ave.'      '9th Fl'   
205                   Ai Media Group'     '483 10th Avenue'   '4th Floor'   
730                   Brick & Portal'     '626 10th Avenue'         '#5E'   
3591                 Ogilvy & Mather'     '636 11th avenue'           NaN   
5509                         Vendedy'   '253-41 149 Avenue'           NaN   
893                  CareerThesaurus'     '65 15th Street '         '#2F'   
4185                    Refined Hype'         '587 17th st'           NaN   
1315      Data Conversion Laboratory'  '61-18 190th Street'   'Suite 205'   
3188                     Method Mill'        '240 1st Ave '    'Suite 1D'   
3772                       Patentory'  ' 214 W 29th Street'           NaN   
5345                   Trust Metrics'          '45 21st St'         '404'   
...                               ...                   ...           ...   
1529                         EYEBEAM'         '34 35th St '   '5th Floor'   
1393                 Digital Brewery'       '36-27 36th St'           NaN   
330                            Arc90'         '747 3rd Ave'  '30th Floor'   
2702                        Kindling'         '747 3rd Ave'  '30th Floor'   
3757              Pariveda Solutions'         '630 3rd Ave'        'Fl 6'   
1091                       Cologuard'         '882 3rd Ave'           NaN   
5850                           Zenly'        '708 3rd Ave.'   '6th Floor'   
3402                    NYC Resistor'       '87 3rd Avenue'   '4th Floor'   
6209                      reMADE LLC'      '232 3rd Street'        'F201'   
362                    ArtistUpgrade'   '34-27 41st Street'           NaN   
110   ALTANOVA Energy+Sustainability'       '11-05 44th Dr'           NaN   
3570                         OKFocus'     '11-11 44th Road'           NaN   

                    City Address_Num         Add_StName  
4978          'New York'         166      Mercer Street  
2171          'New York'         475          10th Ave.  
205           'New York'         483        10th Avenue  
730           'New York'         626        10th Avenue  
3591          'New York'         636        11th avenue  
5509          'Rosedale'      253-41         149 Avenue  
893           'Brooklyn'          65       15th Street   
4185          'New York'         587            17th st  
1315     'Fresh Meadows'       61-18       190th Street  
3188          'New York'         240           1st Ave   
3772          'New York'              214 W 29th Street  
5345          'New York'          45            21st St  
...                  ...         ...                ...  
1529          'Brooklyn'          34           35th St   
1393  'Long Island City'       36-27            36th St  
330           'New York'         747            3rd Ave  
2702          'New York'         747            3rd Ave  
3757          'New York'         630            3rd Ave  
1091          'Brooklyn'         882            3rd Ave  
5850          'New York'         708           3rd Ave.  
3402          'Brooklyn'          87         3rd Avenue  
6209          'Brooklyn'         232         3rd Street  
362   'Long Island City'       34-27        41st Street  
110   'Long Island City'       11-05            44th Dr  
3570  'Long Island City'       11-11          44th Road  

[30 rows x 6 columns]

In [47]:
gb.Add_StName.value_counts()

Broadway                  165
Madison Avenue             25
Fifth Avenue               24
Varick Street              21
Park Avenue South          20
Jay Street                 17
Avenue of the Americas     17
5th Avenue                 16
Water Street               10
Madison Ave                10
5th Ave                    10
Park Avenue                 9
                         ... 
East 92nd Street            1
Piermont Plaza              1
23rd Ave                    1
West St                     1
W 37th                      1
W. 38th St.                 1
West 31st Street            1
28th Street                 1
lafayette street            1
Sullivan St                 1
rider avenue                1
W 57th Street               1
Name: Add_StName, Length: 700, dtype: int64

In [59]:
def Add_StRe(x): 
    name = x
    if ('Broadway' in x):
        name = 'Broadway'
    if (x == '5th Ave') or (x=='Fifth Avenue'):
        name = '5th Avenue'
    if (x== 'Madison Ave') or (x== 'Madison Ave.'):
        name = 'Madison Avenue'
    if (x== 'Jay St.') or (x== 'Jay St') or (x=='Jay st.'):
        name = 'Jay Street'
    if ('Varick' in x):
        name = 'Varick Street'
    if (x== 'Water St'):
        name = 'Water Street'
    return name


In [66]:
gb["Add_StNew"] = gb["Add_StName"].apply(lambda row: Add_StRe(str(row)))
gb.head()

Company Name               Address     Address2        City  \
4978      Tapjoy, Inc'  '166  Mercer Street'    'Floor 2'  'New York'   
2171    Group 8A, LLC'       '475 10th Ave.'     '9th Fl'  'New York'   
205    Ai Media Group'     '483 10th Avenue'  '4th Floor'  'New York'   
730    Brick & Portal'     '626 10th Avenue'        '#5E'  'New York'   
3591  Ogilvy & Mather'     '636 11th avenue'          NaN  'New York'   

     Address_Num      Add_StName       Add_StNew  
4978         166   Mercer Street   Mercer Street  
2171         475       10th Ave.       10th Ave.  
205          483     10th Avenue     10th Avenue  
730          626     10th Avenue     10th Avenue  
3591         636     11th avenue     11th avenue

In [75]:
gb.Add_StNew.value_counts()

Broadway                  203
5th Avenue                 50
Varick Street              42
Madison Avenue             37
Jay Street                 29
Park Avenue South          20
Avenue of the Americas     17
Water Street               11
Park Avenue                 9
Third Avenue                9
Main Street                 8
Lafayette Street            7
                         ... 
Morton Street               1
Jefferson St.               1
W 86TH ST                   1
East 92nd Street            1
Morningside Drive           1
Piermont Plaza              1
23rd Ave                    1
W 37th                      1
West 30th street            1
West 31st Street            1
28th Street                 1
W 57th Street               1
Name: Add_StNew, Length: 658, dtype: int64

In [68]:
gb.City.value_counts()

'New York'                         1033
'Brooklyn'                          156
'Staten Island'                      10
'Long Island City'                   10
'Bronx'                               9
'Queens'                              6
'New York City'                       5
'New York '                           4
'NY'                                  2
'Rosedale'                            1
'RUSTON'                              1
'New Rochelle'                        1
                                   ... 
'Forest Hills'                        1
'Great Neck '                         1
'Larchmont'                           1
'Washington, D.C.'                    1
'Redmond'                             1
'Portland'                            1
'Fresh Meadows'                       1
'Floral Park'                         1
'http://www.8pathsolutions.com/       1
'BROOKLYN'                            1
'Astoria'                             1
'Greenvale'                           1


In [71]:
def City_Re(x): 
    name = x
    if ('New York' in x):
        name = 'New York'
    return name

In [72]:
gb["CityNew"] = gb["City"].apply(lambda row: City_Re(str(row)))
gb.head()

Company Name               Address     Address2        City  \
4978      Tapjoy, Inc'  '166  Mercer Street'    'Floor 2'  'New York'   
2171    Group 8A, LLC'       '475 10th Ave.'     '9th Fl'  'New York'   
205    Ai Media Group'     '483 10th Avenue'  '4th Floor'  'New York'   
730    Brick & Portal'     '626 10th Avenue'        '#5E'  'New York'   
3591  Ogilvy & Mather'     '636 11th avenue'          NaN  'New York'   

     Address_Num      Add_StName       Add_StNew   CityNew  
4978         166   Mercer Street   Mercer Street  New York  
2171         475       10th Ave.       10th Ave.  New York  
205          483     10th Avenue     10th Avenue  New York  
730          626     10th Avenue     10th Avenue  New York  
3591         636     11th avenue     11th avenue  New York

In [74]:
gb.CityNew.value_counts()

New York                           1042
'Brooklyn'                          156
nan                                  58
'Long Island City'                   10
'Staten Island'                      10
'Bronx'                               9
'Queens'                              6
'NY'                                  2
'Brooklyn '                           1
'http://www.twiagemed.com             1
'NEW YORK'                            1
'Melville'                            1
                                   ... 
'Greenvale'                           1
'Washington, D.C.'                    1
'Redmond'                             1
'Portland'                            1
'Larchmont'                           1
'Floral Park'                         1
'BROOKLYN'                            1
'Astoria'                             1
'Rosedale'                            1
'Glen Cove'                           1
'http://www.8pathsolutions.com/       1
'Great Neck '                         1


In [412]:
gb[(gb['Add_StName'].str.contains('Varick', re.IGNORECASE, na=False))]

Company Name  \
154                             AdColony'   
4771                             Spotted'   
159                             Adafruit'   
5177                         The Whistle'   
519                            Beer2Buds'   
1706                        F&T Creative'   
5555                              Visify'   
3770                           Passenger'   
138                    Acceleron Digital'   
2486                        Intent Media'   
4627                         SmokeClouds'   
3396                      NY Tech Meetup'   
3109                    Matter Unlimited'   
764                         BrunchCritic'   
108                                 AKQA'   
293                           Anvil+Gear'   
1249              Cut On Your Bias, Inc.'   
804                   BuzzBuzzHome Corp.'   
2322                    Honest Buildings'   
3470               New York Public Radio'   
...                                   ...   
475             Bandwagon (Weeels, Inc.)'   
2970                            Luminate'   
4952                     Tactile Finance'   
4849                           StyleSage'   
6179                              onkaro'   
1321                            Datacoup'   
3811                       PeoplePerHour'   
6205                                quip'   
1817  Find Your Lobster (3DegreesNation)'   
2993           MAP Digital: MetaMeetings'   
710                       Brand Networks'   
4449            SecureSafe (LifeEnsured)'   
3524                          NoiseTrade'   
2385                  I'm In The Kitchen'   
17                     1800Postcards.com'   
3704                       PBJ Marketing'   
5442                            Unmetric'   
2301                     Hire an Esquire'   
424                         Awl & Sundry'   
5940                      charity: water'   

                                               Address  \
154                                       '175 Varick'   
4771                                   '137 Varick St'   
159               '150 Varick St, New York, NY, 10013'   
5177                                   '175 Varick St'   
519                                    '175 Varick St'   
1706                                   '175 Varick St'   
5555                                  '200 Varick St '   
3770                                  '180 Varick St '   
138                               '131 Varick St #917'   
2486                              '180 Varick St #936'   
4627                               '175 Varick St 409'   
3396     '175 Varick St 4th Floor New York, NY 10014 '   
3109                                  '175 Varick St.'   
764                                   '175 Varick St.'   
108                                   '175 Varick St.'   
293                                   '175 Varick St.'   
1249                                  '131 Varick St.'   
804                                   '175 Varick St.'   
2322                       '137 Varick St., 2nd Floor'   
3470                                  '160 Varick St.'   
...                                                ...   
475                                '137 Varick Street'   
2970                               '175 Varick Street'   
4952                               '137 Varick Street'   
4849  '137 Varick Street, 2nd FL.  New York, NY 10013'   
6179                               '175 Varick Street'   
1321                               '175 Varick Street'   
3811                               '175 Varick Street'   
6205                               '175 Varick Street'   
1817                               '175 Varick Street'   
2993                               '180 Varick Street'   
710                                '175 Varick Street'   
4449                               '160 Varick Street'   
3524                               '175 Varick Street'   
2385                               '175 Varick Street'   
17                                 '121 Varick Str

In [77]:
xy = gb[['Company Name', 'Address', 'Address2','CityNew','Address_Num','Add_StNew']].copy()
xy.head()

Company Name               Address     Address2   CityNew  \
4978      Tapjoy, Inc'  '166  Mercer Street'    'Floor 2'  New York   
2171    Group 8A, LLC'       '475 10th Ave.'     '9th Fl'  New York   
205    Ai Media Group'     '483 10th Avenue'  '4th Floor'  New York   
730    Brick & Portal'     '626 10th Avenue'        '#5E'  New York   
3591  Ogilvy & Mather'     '636 11th avenue'          NaN  New York   

     Address_Num       Add_StNew  
4978         166   Mercer Street  
2171         475       10th Ave.  
205          483     10th Avenue  
730          626     10th Avenue  
3591         636     11th avenue

In [78]:
xy.Add_StNew.value_counts()

Broadway                  203
5th Avenue                 50
Varick Street              42
Madison Avenue             37
Jay Street                 29
Park Avenue South          20
Avenue of the Americas     17
Water Street               11
Park Avenue                 9
Third Avenue                9
Main Street                 8
Lafayette Street            7
                         ... 
Morton Street               1
Jefferson St.               1
W 86TH ST                   1
East 92nd Street            1
Morningside Drive           1
Piermont Plaza              1
23rd Ave                    1
W 37th                      1
West 30th street            1
West 31st Street            1
28th Street                 1
W 57th Street               1
Name: Add_StNew, Length: 658, dtype: int64

### Keeping only New York

In [83]:
xy = xy[(xy['CityNew']=='New York')]
xy.CityNew.value_counts()

New York    1042
Name: CityNew, dtype: int64

### Counting the no. of companies

In [99]:
pq = xy.groupby(['Add_StNew','Address_Num'])[['Company Name']].count()

In [100]:
pq.head()

Company Name
Add_StNew                   Address_Num              
 Mercer Street              166                     1
10th Ave.                   475                     1
10th Avenue                 483                     1
                            626                     1
11th avenue                 636                     1
17th st                     587                     1
1st Ave                     240                     1
214 W 29th Street                                   1
21st St                     45                      1
28th street                 110E                    1
2nd Ave Apt 4FN             1727                    1
3rd Ave                     630                     1
...                                               ...
west 13th st                425                     1
west 19th street            33                      1
west 20th st                17                      1
west 24th St.               27                      1
west 26th                   122                     1
west 28th st                150                     1
west 31st street            119                     1
west 38th street            262                     1
west 39th street            28                      1
west 42nd street            330                     1
west 43rd street suite 1404 311                     1
west 57th street            555                     0

[819 rows x 1 columns]

In [127]:
pq_new = pq.sort_values(by='Company Name', ascending= False).reset_index()
pq_new.rename(columns={'Company Name': 'Company Count'},inplace=True)

In [128]:
pq_new

Add_StNew Address_Num  Company Count
0       Varick Street         175             23
1            Broadway         902             22
2            Broadway         560              7
3            Broadway         222              7
4      Madison Avenue          79              6
5            Broadway        1133              6
6       Varick Street         137              6
7          5th Avenue         244              6
8            Broadway         447              6
9            Broadway         394              6
10           Broadway         636              5
11         5th Avenue         349              5
12           Broadway         594              5
13   West 26th Street         122              4
14         5th Avenue          85              4
15           Broadway          29              4
16           Broadway          11              4
17           Broadway         379              4
18         5th Avenue         110              4
19           Broadway         611              4
..                ...         ...            ...
799  East 25th Street         110              1
800  East 26th Street          11              1
801  East 27th Street         116              1
802        East 28 St         110              1
803  East 28th Street          28              1
804  East 29th Street         217              1
805  East 30th Street         333              1
806      East 32nd St          31              1
807     East 32nd St.           6              1
808  East 32nd Street          31              1
809  East 32nd Street           6              1
810     East 33rd St.          33              1
811  East 33rd Street          10              1
812  East 34th Street          10              1
813  East 38th Street          10              1
814       East 39 St.         142              1
815      East 39th St          10              1
816      East 41st St          18              1
817  East 42nd Street          60              1
818  west 57th street         555              0

[819 rows x 3 columns]

In [372]:
pq_new.Add_StNew.value_counts()

Broadway                       85
5th Avenue                     28
Madison Avenue                 24
Park Avenue South              15
Avenue of the Americas         14
Lafayette Street                7
Third Avenue                    7
Varick Street                   7
Fifth Ave                       6
Park Avenue                     6
7th Ave                         5
Lexington Ave                   5
Grand Street                    5
East 23rd Street                5
Water Street                    5
West 18th Street                5
West 38th Street                4
West 26th Street                4
Park Ave South                  4
W 38th Street                   4
                               ..
W 14th St                       1
w 13th St                       1
E 57th St #1026                 1
E 27th St                       1
West 31st Street                1
85th St                         1
W 22 Street                     1
83rd Street                     1
2nd Ave Apt 4F

#### Making seperate DFs for each Add_StNew by grouping 200 adjacent Addresses on that Street/Avenue.  These addresses are within a .8miles distance

In [386]:
VarickSt_df = pq_new[(pq_new['Add_StNew'].str.contains('Varick', re.IGNORECASE, na=False))]
VarickSt = VarickSt_df.groupby(['AddMerge'])[['Company Count']].sum()
 #xy.groupby(['Add_StNew','Address_Num'])[['Company Name']].count()
VarickSt

Company Count
AddMerge                            
100-200 Varick Street             40

In [330]:
FifthAvenue_df = pq_new[(pq_new['Add_StNew'].str.contains('5th Avenue', re.IGNORECASE, na=False))]
FifthAvenue_df

Add_StNew Address_Num  Company Count AddMerge
7              5th Avenue         244              6         
11             5th Avenue         349              5         
14             5th Avenue          85              4         
18             5th Avenue         110              4         
87             5th Avenue         350              2         
101            5th Avenue         565              2         
542            5th Avenue         545              1         
543            5th Avenue         281              1         
544            5th Avenue         286              1         
545            5th Avenue         298              1         
546            5th Avenue         303              1         
547            5th Avenue         307              1         
548            5th Avenue         417              1         
549            5th Avenue         433              1         
550            5th Avenue         437              1         
551            5th Avenue         501              1         
552            5th Avenue         610              1         
554            5th Avenue         666              1         
555            5th Avenue         680              1         
556            5th Avenue          78              1         
557  5th Avenue 6th Floor         554              1         
558  5th Avenue 7th Floor          89              1         
563            5th Avenue         261              1         
564            5th Avenue         226              1         
565            5th Avenue         137              1         
566            5th Avenue         126              1         
582            5th Avenue         100              1         
583            5th Avenue         104              1         
584            5th Avenue         115              1         
585            5th Avenue         120              1

In [331]:
def Add_Merge(x):
    a= ''
    for i in range(1,1400,200):
        if x in range(i,i+200):
            a= str(i)+"-"+str(i+200)+" "+'5th Avenue'     
    return a  

In [332]:
FifthAvenue_df["AddMerge"] = FifthAvenue_df.apply(lambda row: Add_Merge(int(row['Address_Num'])), axis=1)

/Users/sdutta/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [333]:
FifthAvenue_df

Add_StNew Address_Num  Company Count            AddMerge
7              5th Avenue         244              6  201-401 5th Avenue
11             5th Avenue         349              5  201-401 5th Avenue
14             5th Avenue          85              4    1-201 5th Avenue
18             5th Avenue         110              4    1-201 5th Avenue
87             5th Avenue         350              2  201-401 5th Avenue
101            5th Avenue         565              2  401-601 5th Avenue
542            5th Avenue         545              1  401-601 5th Avenue
543            5th Avenue         281              1  201-401 5th Avenue
544            5th Avenue         286              1  201-401 5th Avenue
545            5th Avenue         298              1  201-401 5th Avenue
546            5th Avenue         303              1  201-401 5th Avenue
547            5th Avenue         307              1  201-401 5th Avenue
548            5th Avenue         417              1  401-601 5th Avenue
549            5th Avenue         433              1  401-601 5th Avenue
550            5th Avenue         437              1  401-601 5th Avenue
551            5th Avenue         501              1  401-601 5th Avenue
552            5th Avenue         610              1  601-801 5th Avenue
554            5th Avenue         666              1  601-801 5th Avenue
555            5th Avenue         680              1  601-801 5th Avenue
556            5th Avenue          78              1    1-201 5th Avenue
557  5th Avenue 6th Floor         554              1  401-601 5th Avenue
558  5th Avenue 7th Floor          89              1    1-201 5th Avenue
563            5th Avenue         261              1  201-401 5th Avenue
564            5th Avenue         226              1  201-401 5th Avenue
565            5th Avenue         137              1    1-201 5th Avenue
566            5th Avenue         126              1    1-201 5th Avenue
582            5th Avenue         100              1    1-201 5th Avenue
583            5th Avenue         104              1    1-201 5th Avenue
584            5th Avenue         115              1    1-201 5th Avenue
585            5th Avenue         120              1    1-201 5th Avenue

In [390]:
FifthAvenue= FifthAvenue_df.groupby(['AddMerge'])[['Company Count']].sum()
FifthAvenue

Company Count
AddMerge                         
1-201 5th Avenue               16
201-401 5th Avenue             20
401-601 5th Avenue              8
601-801 5th Avenue              3

In [351]:
Broadway_df = pq_new[(pq_new['Add_StNew'].str.contains('Broadway', re.IGNORECASE, na=False))].reset_index()
Broadway_df

index Add_StNew Address_Num  Company Count AddMerge
0       1  Broadway         902             22      nnj
1       2  Broadway         560              7         
2       3  Broadway         222              7         
3       5  Broadway        1133              6         
4       8  Broadway         447              6         
5       9  Broadway         394              6         
6      10  Broadway         636              5         
7      12  Broadway         594              5         
8      15  Broadway          29              4         
9      16  Broadway          11              4         
10     17  Broadway         379              4         
11     19  Broadway         611              4         
12     22  Broadway         291              3         
13     23  Broadway         632              3         
14     25  Broadway         195              3         
15     27  Broadway         770              3         
16     28  Broadway        1123              3         
17     29  Broadway        1440              3         
18     30  Broadway         412              3         
19     32  Broadway         568              3         
..    ...       ...         ...            ...      ...
65    488  Broadway        1270              1         
66    489  Broadway        1350              1         
67    490  Broadway        1370              1         
68    491  Broadway        1400              1         
69    492  Broadway         578              1         
70    493  Broadway         599              1         
71    506  Broadway          61              1         
72    520  Broadway         627              1         
73    521  Broadway         628              1         
74    522  Broadway          65              1         
75    523  Broadway         665              1         
76    524  Broadway         670              1         
77    525  Broadway         833              1         
78    526  Broadway         841              1         
79    527  Broadway         853              1         
80    528  Broadway         875              1         
81    529  Broadway          90              1         
82    530  Broadway         907              1         
83    531  Broadway         915              1         
84    532  Broadway         920              1         

[85 rows x 5 columns]

In [358]:
def Add_Merge(x):
    a= ''
    for i in range(1,2000,200):
        if x in range(i,i+200):
            a= str(i)+"-"+str(i+200)+" "+'Broadway'     
    return a  

In [359]:
Broadway_df = Broadway_df[~Broadway_df['Address_Num'].isin(['101-104','One','14W.23rd','Hudson','1st','Union','98th','italy','41-43','NeueHouse','Montgromery','Brooklyn','--','Block'])]

 ### Issue with entry in Address_Num field of this row

In [360]:
Broadway_df=Broadway_df.drop(Broadway_df.index[60])

In [361]:
Broadway_df["AddMerge"] = Broadway_df.apply(lambda row: Add_Merge(int(row['Address_Num'])), axis=1)

In [362]:
Broadway_df

index Add_StNew Address_Num  Company Count            AddMerge
0       1  Broadway         902             22   801-1001 Broadway
1       2  Broadway         560              7    401-601 Broadway
2       3  Broadway         222              7    201-401 Broadway
3       5  Broadway        1133              6  1001-1201 Broadway
4       8  Broadway         447              6    401-601 Broadway
5       9  Broadway         394              6    201-401 Broadway
6      10  Broadway         636              5    601-801 Broadway
7      12  Broadway         594              5    401-601 Broadway
8      15  Broadway          29              4      1-201 Broadway
9      16  Broadway          11              4      1-201 Broadway
10     17  Broadway         379              4    201-401 Broadway
11     19  Broadway         611              4    601-801 Broadway
12     22  Broadway         291              3    201-401 Broadway
13     23  Broadway         632              3    601-801 Broadway
14     25  Broadway         195              3      1-201 Broadway
15     27  Broadway         770              3    601-801 Broadway
16     28  Broadway        1123              3  1001-1201 Broadway
17     29  Broadway        1440              3  1401-1601 Broadway
18     30  Broadway         412              3    401-601 Broadway
19     32  Broadway         568              3    401-601 Broadway
..    ...       ...         ...            ...                 ...
65    488  Broadway        1270              1  1201-1401 Broadway
66    489  Broadway        1350              1  1201-1401 Broadway
67    490  Broadway        1370              1  1201-1401 Broadway
68    491  Broadway        1400              1  1201-1401 Broadway
69    492  Broadway         578              1    401-601 Broadway
70    493  Broadway         599              1    401-601 Broadway
71    506  Broadway          61              1      1-201 Broadway
72    520  Broadway         627              1    601-801 Broadway
73    521  Broadway         628              1    601-801 Broadway
74    522  Broadway          65              1      1-201 Broadway
75    523  Broadway         665              1    601-801 Broadway
76    524  Broadway         670              1    601-801 Broadway
77    525  Broadway         833              1   801-1001 Broadway
78    526  Broadway         841              1   801-1001 Broadway
79    527  Broadway         853              1   801-1001 Broadway
80    528  Broadway         875              1   801-1001 Broadway
81    529  Broadway          90              1      1-201 Broadway
82    530  Broadway         907              1   801-1001 Broadway
83    531  Broadway         915              1   801-1001 Broadway
84    532  Broadway         920              1   801-1001 Broadway

[83 rows x 5 columns]

In [396]:
Broadway=Broadway_df.groupby(['AddMerge'])[['Company Count']].sum()
Broadway

Company Count
AddMerge                         
                                3
1-201 Broadway                 27
1001-1201 Broadway             14
1201-1401 Broadway              8
1401-1601 Broadway              8
1601-1801 Broadway              2
201-401 Broadway               27
401-601 Broadway               43
601-801 Broadway               21
801-1001 Broadway              31

In [403]:
Broadway= Broadway[1:]
Broadway

Company Count
AddMerge                         
1001-1201 Broadway             14
1201-1401 Broadway              8
1401-1601 Broadway              8
1601-1801 Broadway              2
201-401 Broadway               27
401-601 Broadway               43
601-801 Broadway               21
801-1001 Broadway              31

In [397]:
Madison_df = pq_new[(pq_new['Add_StNew'].str.contains('Madison', re.IGNORECASE, na=False))]

def Add_Merge(x):
    a= ''
    for i in range(1,1400,200):
        if x in range(i,i+200):
            a= str(i)+"-"+str(i+200)+" "+'Madison Avenue'     
    return a  

Madison_df = Madison_df[~Madison_df['Address_Num'].isin(['101-104','One','14W.23rd','Hudson','1st','Union','98th','italy','41-43','NeueHouse','Montgromery','Brooklyn','--','Block'])]

Madison_df["AddMerge"] = Madison_df.apply(lambda row: Add_Merge(int(row['Address_Num'])), axis=1)

Madison=Madison_df.groupby(['AddMerge'])[['Company Count']].sum()
Madison

Company Count
AddMerge                             
1-201 Madison Avenue               20
201-401 Madison Avenue             15
401-601 Madison Avenue              3
601-801 Madison Avenue              2

In [370]:
Jay_df = pq_new[(pq_new['Add_StNew'].str.contains('Jay Street', re.IGNORECASE, na=False))]
Jay_df

Add_StNew Address_Num  Company Count AddMerge
651  Jay Street          10              1         
652  Jay Street          20              1         
653  Jay Street          68              1

In [368]:
def Add_Merge(x):
    a= 'Jay Street'
    for i in range(1,1400,200):
        if x in range(i,i+200):
            a= str(i)+"-"+str(i+200)+" "+'Jay Street'     
    return a  

#Jay_df = Jay_df[~Jay_df['Address_Num'].isin(['101-104','One','14W.23rd','Hudson','1st','Union','98th','italy','41-43','NeueHouse','Montgromery','Brooklyn','--','Block'])]

Jay_df["AddMerge"] =Jay_df.apply(lambda row: Add_Merge(int(row['Address_Num'])), axis=1)

Jay_df.groupby(['AddMerge'])[['Company Count']].sum()

/Users/sdutta/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:12: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if sys.path[0] == '':


Company Count
AddMerge                       
1-201 Jay Street              3

In [374]:
PAVS_df = pq_new[(pq_new['Add_StNew'].str.contains('Park Avenue South', re.IGNORECASE, na=False))]

In [405]:
def Add_Merge(x):
    
    for i in range(1,1400,200):
        if x in range(i,i+200):
            a= str(i)+"-"+str(i+200)+" "+'Park Avenue South'    
    return a  

#Jay_df = Jay_df[~Jay_df['Address_Num'].isin(['101-104','One','14W.23rd','Hudson','1st','Union','98th','italy','41-43','NeueHouse','Montgromery','Brooklyn','--','Block'])]

PAVS_df["AddMerge"] =PAVS_df .apply(lambda row: Add_Merge(int(row['Address_Num'])), axis=1)

PAVS = PAVS_df.groupby(['AddMerge'])[['Company Count']].sum()
PAVS

/Users/sdutta/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  # Remove the CWD from sys.path while we load stuff.


Company Count
AddMerge                                
201-401 Park Avenue South             11
401-601 Park Avenue South             11

In [377]:
AvAm_df = pq_new[(pq_new['Add_StNew'].str.contains('Avenue of the Americas', re.IGNORECASE, na=False))]

In [407]:
def Add_Merge(x):
    
    for i in range(1,1400,200):
        if x in range(i,i+200):
            a= str(i)+"-"+str(i+200)+" "+'Avenue of the Americas'    
    return a  

#Jay_df = Jay_df[~Jay_df['Address_Num'].isin(['101-104','One','14W.23rd','Hudson','1st','Union','98th','italy','41-43','NeueHouse','Montgromery','Brooklyn','--','Block'])]

AvAm_df["AddMerge"] =AvAm_df.apply(lambda row: Add_Merge(int(row['Address_Num'])), axis=1)

AvAm=AvAm_df.groupby(['AddMerge'])[['Company Count']].sum()
AvAm

/Users/sdutta/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  # Remove the CWD from sys.path while we load stuff.


Company Count
AddMerge                                       
1-201 Avenue of the Americas                  4
1001-1201 Avenue of the Americas              5
1201-1401 Avenue of the Americas              7
601-801 Avenue of the Americas                1

In [379]:
WaterSt_df = pq_new[(pq_new['Add_StNew'].str.contains('Water Street', re.IGNORECASE, na=False))]

In [387]:
def Add_Merge(x):
    
    for i in range(1,1400,200):
        if x in range(i,i+200):
            a= str(i)+"-"+str(i+200)+" "+'Water Street'    
    return a  

#Jay_df = Jay_df[~Jay_df['Address_Num'].isin(['101-104','One','14W.23rd','Hudson','1st','Union','98th','italy','41-43','NeueHouse','Montgromery','Brooklyn','--','Block'])]

WaterSt_df["AddMerge"] =WaterSt_df.apply(lambda row: Add_Merge(int(row['Address_Num'])), axis=1)

WaterSt= WaterSt_df.groupby(['AddMerge'])[['Company Count']].sum()
WaterSt

/Users/sdutta/anaconda/lib/python3.6/site-packages/ipykernel_launcher.py:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  # Remove the CWD from sys.path while we load stuff.


Company Count
AddMerge                           
1-201 Water Street                2
201-401 Water Street              3

### Merging all the DFs of the main Streets

In [408]:
final = pd.concat([Broadway,Madison,FifthAvenue,PAVS,VarickSt,AvAm,WaterSt])
final

Company Count
AddMerge                                       
1001-1201 Broadway                           14
1201-1401 Broadway                            8
1401-1601 Broadway                            8
1601-1801 Broadway                            2
201-401 Broadway                             27
401-601 Broadway                             43
601-801 Broadway                             21
801-1001 Broadway                            31
1-201 Madison Avenue                         20
201-401 Madison Avenue                       15
401-601 Madison Avenue                        3
601-801 Madison Avenue                        2
1-201 5th Avenue                             16
201-401 5th Avenue                           20
401-601 5th Avenue                            8
601-801 5th Avenue                            3
201-401 Park Avenue South                    11
401-601 Park Avenue South                    11
100-200 Varick Street                        40
1-201 Avenue of the Americas                  4
1001-1201 Avenue of the Americas              5
1201-1401 Avenue of the Americas              7
601-801 Avenue of the Americas                1
1-201 Water Street                            2
201-401 Water Street                          3

In [411]:
final.sort_values(by='Company Count', ascending =False).head(10)

Company Count
AddMerge                             
401-601 Broadway                   43
100-200 Varick Street              40
801-1001 Broadway                  31
201-401 Broadway                   27
601-801 Broadway                   21
201-401 5th Avenue                 20
1-201 Madison Avenue               20
1-201 5th Avenue                   16
201-401 Madison Avenue             15
1001-1201 Broadway                 14

401-601 Broadway - .6mile - Prince St. Station NQRW, Broadway-Lafayette St BDFM, Canal St. NQRW

100-200 Varick Street - .3mile - Houston St. 125 , CANAL ST. 125

801-1001 Broadway - .7mile - 23rd St. NQRW

201-401 Broadway - .7mile - Canal St., Fulton St. , Path WTC

601-801 Broadway -.6miles - Prince St. Station NQRW,Broadway-Lafayette St BDFM, Bleeker St

201-401 5th Avenue - .6 miles -34th St Herald Square

1-201 Madison Avenue -.6 miles - E 28th St. NQR, 33rd St, 34th St Herald Square

201-401 Madison Avenue - .6miles - Grand Central Terminal, 42 St. Bryant Park

1001-1201 Broadway -.2miles - E 28th St NQRW, E 23rd St.

1-201 5th Avenue - .8miles, E 23rd St. 466, 3 Av, 14th St. Path Station